### PR TITLE
rp2: Add SD card + FAT support.

### PR DESCRIPTION
**Pull request by StereoRocker**
manifest.py: Added sdcard module.
mpconfigport.h: Added FAT driver and configuration.

### CODE REVIEW

**Comment by robert-hh** [comment]
That looks a littel bit simpler than my changes. I had a typo there. But I wonder how you come along without defining the port-specific function get_fattime()?

**Comment by StereoRocker** [comment]
Thanks @robert-hh - my initial attempt borrowed the get_fattime() code from the esp32 port. But I'm fairly sure I'm able to omit it now because I defined MICROPY_FATFS_NORTC in mpconfigport.h

**Comment by robert-hh** [comment]
I went a similar way but made a get_fattime() function, because it's there. See: https://github.com/robert-hh/micropython/blob/rp2_fat/ports/rp2/fatfs_port.c

**Comment by StereoRocker** [comment]
That's really great! Is it perhaps worth including RTC support for the fatfs port, alongside RTC support in the machine module? To my knowledge, the RTC hasn't yet been ported to the machine module. I'm thinking if we can't set the RTC yet, there's little value in it being used in fatfs.

**Comment by robert-hh** [comment]
The time functions like time.localtime() and time.gmtime() already use the rtc. It in intialized in main.c. What is missing is the API to set date & time. So I think it's ok to prepare get_fattime() accordingly. If the rtc module comes along allowing to set the time, the other components are prepared.

**Comment by StereoRocker** [comment]
Great, that makes sense. I'll include your implementation of fatfs_port.c, and remove MICROPY_FATFS_NORTC in this PR.

Maybe I'll look at the RTC APIs today, then, if that's all that's missing!

**Comment by dpgeorge** [comment]
> Maybe I'll look at the RTC APIs

The main issue there, and why it's not already in place, is that there currently is not a good standard for setting the RTC in MicroPython.  So the first thing to do is define this standard.

**Comment by StereoRocker** [comment]
> So the first thing to do is define this standard.

Is this already being discussed somewhere?

As things already stand, I see the RTC is set in other ports by calling rtc.init(), and it could be implemented the same way for RP2040. Equally, if there's already a discussion about how to set RTC for all ports, it could be worth awaiting the conclusion and resulting standard before implementing for RP2040.

**Comment by dpgeorge** [comment]
I'm happy to merge this to enable FAT support.

But does it make sense to freeze in the sdcard.py driver as well?  That's something that can be quite easily put on the filesystem if the user wants that feature.

File: ports/rp2/fatfs_port.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "lib/oofatfs/ff.h"
+#include "hardware/rtc.h"
+
+MP_WEAK DWORD get_fattime(void) {
+    datetime_t t;
+    rtc_get_datetime(&t);
+    return ((2000 + t.year - 1980) << 25) | ((t.month) << 21) | ((t.day) << 16) | ((t.hour) << 11) | ((t.min) << 5) | (t.sec / 2);
+    // Dummy date: Jan 1st, 2018 at midnight. Not sure what timezone.
+    // return ((2018 - 1980) << 25) | ((1) << 21) | ((1) << 16) | ((0) << 11) | ((0) << 5) | (0 / 2);
```
Comment (dpgeorge) [inline ports/rp2/fatfs_port.c:35]: these 2 lines of comments aren't needed

**Comment by StereoRocker** [comment]
Agreed, sdcard.py can be easily added by the user. I'll remove the freezing for that. 

I'll also remove the comments as you've advised.

**Comment by dpgeorge** [comment]
Thanks for updating, squashed and merged in 8610bababed3703f52c57f7e7b575c9d50be38a4

**Pull request closed without merging**